src/Models/RaspberryPiDevice.ts

```diff
@@ -42,6 +42,32 @@ export class RaspberryPiDevice extends ContainerDeviceBase {
         DeviceType.Raspberry_Pi, templateFilesInfo);
   }
 
+  private async getBinaryFileName(): Promise<string|undefined> {
+    // Parse binary name from CMakeLists.txt file
+    const cmakeFilePath = path.join(this.projectFolder, 'CMakeLists.txt');
+    if (!await FileUtility.fileExists(ScaffoldType.Workspace, cmakeFilePath)) {
+      return;
+    }
+    const getBinaryFileNameCmd = `cat ${
+        cmakeFilePath} | grep 'set(binary_name' | cut -d ' ' -f2 | sed -e 's/^"//' -e 's/"$//' | tr -d '\n'`;
+
+    const binaryName = await execute(getBinaryFileNameCmd);
+    return binaryName;
+  }
+
+  private async enableBinaryExecutability(
+      ssh: sdk.SSH, binaryName: string|undefined) {

```

Comment:
Per the method name, Suggest ensuring binaryName is a string and remove undefined here.

If a undefined binaryName is a ignorable case, please check in the caller, otherwise throw error for this case.

Reply:
Currently we change binary file executability when we can find the binary file in the correct place. Since we expose quite large extensible freedom to user, we will just ignore if we cannot find the binary file in the supposed place.
I will move the logic outside of function `enableBinaryExecutability` and add comment to describe the situation.